pkg/kubectl/cmd/run.go

```diff
@@ -276,23 +275,19 @@ func RunRun(f cmdutil.Factory, cmdIn io.Reader, cmdOut, cmdErr io.Writer, cmd *c
 
 	params["env"] = cmdutil.GetFlagStringArray(cmd, "env")
 
-	allErrs := []error{}
 	var runObjectMap = map[string]*RunObject{}
 	runObject, err := createGeneratedObject(f, cmd, generator, names, params, cmdutil.GetFlagString(cmd, "overrides"), namespace)
 	if err != nil {
-		allErrs = append(allErrs, err)
-	} else {
-		runObjectMap[generatorName] = runObject
+		return err
 	}
+	runObjectMap[generatorName] = runObject
 	if cmdutil.GetFlagBool(cmd, "expose") {
 		serviceGenerator := cmdutil.GetFlagString(cmd, "service-generator")
 		if len(serviceGenerator) == 0 {
 			return cmdutil.UsageErrorf(cmd, "No service generator specified")
 		}
 		serviceRunObject, err := generateService(f, cmd, args, serviceGenerator, params, namespace, cmdOut)
-		if err != nil {
-			allErrs = append(allErrs, err)
-		} else {
+		if err == nil {
```

Comment:
We should deal with the error if `err != nil`, rather than ignore it.

Reply:
Good catch. I didn't aware that the `err` was scoped.
I also fixed another problem in the `run` command.